logstash-core/src/main/java/org/logstash/plugins/AliasRegistry.java

```diff
@@ -37,13 +49,91 @@ PluginCoordinate withName(String name) {
         }
     }
 
+    private static class AliasYamlLoader {
+
+        private String extractedHash;
+        private String yaml;
```

Comment:
nit: as a one-off solution that I don't see changing much or in multithreaded situations, I don't necessarily think it is worth spending too much time refining. If you'd like to, then great, and if not, this has a sharp edge that no one will likely hit.

***

In general, using properties as transient local storage is problematic because:
 - it makes the whole object not thread-safe. 
 - the transient state is not cleaned up by GC until the whole object is reaped.

In this particular case, the class is `private` and in its only usage instances are immediately discarded after construction and an immediate method call, so neither issue is particularly pertinent. But code evolves over time and I like being in the habit of avoiding creating sharp edges where possible.

Another way to do this would be to define a tuple-like class and use it to carry all the required state, returning that state tuple from the private method that does the extraction. In this case, if our `parseYamlFile` method were to _return_ a tuple containing the yaml contents and the extracted checksum (say `YamlWithChecksum`?), the state would not need to be persisted to the `AliasYamlLoader` and could instead be a local variable in `AliasYamlLoader#loadAliasesDefinitions` where it is used.

Additionally, the extraction of these properties from a file path could be pulled down into the `YamlFileWithChecksum` class, neatly wrapping all of the functionality about loading the yaml into a single class. Our `AliasYamlLoader` would no longer carry any state so  `AliasYamlLoader#loadAliasesDefinitions` could either become static `AliasYamlLoader::load` or static `AliasRegistry::loadAliasDefinitions`.

~~~
    private static class YamlWithChecksum {

        static YamlWithChecksum load(final String filePath) {
          # read file contents, create new YamlWithChecksum instance with yamlContents and checksumHash
        }

        final String yamlContents;
        final String checksumHash;
        
        private YamlWithChecksum(final String yamlContents, final String checksumHash) {
            this.yamlContents = yamlContents;
            this.checksumHash = checksumHash;
        }
    }
~~~

Reply:
Good point, thanks for it, check commit https://github.com/elastic/logstash/pull/12841/commits/a9cdbc08a1a45c52e50fe30417c1601d767c056f

I've also moved the code to decode the yml to maps and the logic to calculate the hash from content in that tuple class.